src/addRemoveTask.js

```diff
@@ -0,0 +1,44 @@
+export default class AddToDoList {
+  constructor() {
+    this.list = localStorage.getItem('taskList')
+      ? JSON.parse(localStorage.getItem('taskList'))
+      : [];
+  }
+
+  addTask(todo) {
+    this.list.push(todo);
+    localStorage.setItem('taskList', JSON.stringify(this.list));
+  }
+
+  removeTask(todoID) {
+    this.list = this.list.filter((todo) => todo.id !== todoID);
+    this.list.forEach((todo, index) => {
+      todo.index = index + 1;
+    });
+    localStorage.setItem('taskList', JSON.stringify(this.list));
+  }
+
+  editTask(todoId, todoDescription) {
+    this.list = this.list.map((todo) => {
+      if (todo.id === todoId) {
+        return { ...todo, description: todoDescription };
+      }
+      return todo;
+    });
+    localStorage.setItem('taskList', JSON.stringify(this.list));
+  }
+
+  completeTask(todoId, status) {
+    const selected = this.list.findIndex((element) => element.id === todoId);
+    this.list[selected].completed = status;
+    localStorage.setItem('taskList', JSON.stringify(this.list));
+  }
+
+  clearCompletedTask() {
+    this.list = this.list.filter((todo) => !todo.completed);
+    this.list.forEach((todo, index) => {
+      todo.index = index + 1;
+    });
+    localStorage.setItem('taskList', JSON.stringify(this.list));
```

Comment:
[OPTIONAL]

it would be nice if you could make line number 10 a const outside all functions(putting the const at line number 7) and use a small name for it. like
const setLocalStorage = localStorage.setItem('taskList', JSON.stringify(this.list)); and then in your every method you can call this const.

Reply:
Hello @PRATAP-KUMAR I appreciate your review 😄💯. Thank you for going through my work and writing the review it is highly appreciated 👋 
I would like to ask for more clarification on this review. I have tried implementing the above optional review but it causes an error. Kindly help me fix this 
![image](https://user-images.githubusercontent.com/83514256/184859092-9f3125f4-008d-492d-8a2c-5684ddacdb9b.png)
****

Reply:
Hi, @David-Kasilia thanks for replying, I apologize for my [OPTIONAL] suggestion. I have gone through that implementation(const) and it did not work as you mentioned.

I did try to make it a function instead of declaring it as const and called it in each method like this.

````
  setLocalStorage() {
    return localStorage.setItem('taskList', JSON.stringify(this.list));
  }
````

then calling this function in each method like below

````
  addTask(todo) {
    this.list.push(todo);
    this.setLocalStorage();
  }

  removeTask(todoID) {
    this.list = this.list.filter((todo) => todo.id !== todoID);
    this.list.forEach((todo, index) => {
      todo.index = index + 1;
    });
    this.setLocalStorage();
  }
````

Cheers

Happy Coding